src/context/ReduxContext.tsx

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable @typescript-eslint/indent */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useEffect, useReducer } from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api';
+import { User } from '../types/User';
+
+type Action =
+  | { type: 'setTodos'; payload: Todo[] }
+  | { type: 'setUsers'; payload: User }
+  | { type: 'select'; value: string }
+  | { type: 'filter'; value: string }
+  | { type: 'clearQuery' }
+  | {
+      type: 'seeMore';
+      curruntId: number;
+      currentTitile: string;
+      statusComleted: boolean;
+      userId: number;
+    }
+  | { type: 'close' };
+
+interface State {
+  todos: Todo[];
+  user: User | null;
+  select: string;
+  query: string;
+  clearQuery: boolean;
+  seeMore: boolean;
+  userId: number;
+  statusCompleted: boolean;
+  currentTitle: string;
+  currentId: number;
```

Comment:
just save the selectedTodo
```suggestion
selectedTodo: Todo | null
```
if it's null then we don't show the modal, if it's Todo then show it :)